### PR TITLE
Move border-radius custom properties to tokens

### LINKED ISSUE #4611: Move --p-border-radius custom properties to polaris-react

### LINKED ISSUE #4612: Remove --p-text-field-focus-ring-border-radius

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Closes #4612 
Closes #4611 

### WHAT is this pull request doing?

- Removes `--p-text-field-focus-ring-border-radius`
- Moves `--p-border-radius-*` legacy tokens to new token file
- Replaces any custom properties and values related to `border-radius` with tokens where possible

### CODE REVIEW

File: src/tokens/_border.ts
Comment (lgriffee) [inline src/tokens/_border.ts]: Based on my [audit of legacy border token instances](https://github.com/Shopify/polaris-react/issues/4611#issuecomment-982705882) I think we can remove `border-radius-slim` since it's not used in `polaris-react` or `web`. Let me know if that feels in scope for this PR.

**Comment by lgriffee** [comment]
Do we also want to deal with moving other legacy border tokens ([see audit of legacy border token instances for context](https://github.com/Shopify/polaris-react/issues/4611#issuecomment-982705882)) to the new border token file? Or does that feel out of scope for this PR?

File: src/tokens/_border.ts
Comment (alex-page) [reply src/tokens/_border.ts]: I would order this `0.2` `0.4` `0.8` `50%` as that is the roundness increasing. Feel free to remove unused values.

File: src/tokens/_border.ts
Comment (alex-page) [inline src/tokens/_border.ts]: I am curious if we should put radii in a border category? It feels more related to shape or roundness. Interested to know peoples thoughts.

How would this be documented? @sarahill maybe you have  some opinions?

**Review by alex-page — APPROVED** [review]
Awesome work Laura. I added some comments that are not blocking and could be clean up work in the future.

File: src/tokens/_border.ts
Comment (aveline) [reply src/tokens/_border.ts]: I agree it feels more related to shape. But not a blocker.

**Review by aveline — APPROVED** [review]
🎉

File: src/tokens/_border.ts
Comment (aaronccasanova) [reply src/tokens/_border.ts]: Agreed. I think `shape` is an appropriate name for this token group.

File: src/tokens/_legacy-tokens.ts
Comment (aaronccasanova) [inline src/tokens/_legacy-tokens.ts:13]: Thanks for the due diligence in cleaning up these legacy tokens 👍

**Review by aaronccasanova — APPROVED** [review]
🥇

File: src/components/Avatar/Avatar.scss
Comment (alex-page) [inline src/components/Avatar/Avatar.scss]: These might have some conflicts with @kyledurand's PR. We also need to do `--pc-avatar-` so we don't get any nested problems with similar custom properties e.g. `--pc-small-size` for avatar and `--pc-small-size` for text inside it.

File: src/tokens/_shape.ts
Comment (alex-page) [inline src/tokens/_shape.ts]: These should use `px` values like our spacing values. 0.2rem = 2px. If possible we should also follow the same number pattern used there for consistency.

**Review by alex-page — APPROVED** [review]

**Comment by lgriffee** [comment]
Before merging this there are several strange styling issues with a few components that need to be resolved (see Chromatic). @kyledurand and I jammed on a few this afternoon but there's still more to resolve.

File: src/components/RangeSlider/components/DualThumb/DualThumb.tsx
Comment (kyledurand) [inline src/components/RangeSlider/components/DualThumb/DualThumb.tsx:12]: This is a bit of an odd convention but I think makes sense for future proofing. In the future we can apply shared  classes and 🔥 mixins 😄

File: src/components/RangeSlider/components/SingleThumb/SingleThumb.scss
Comment (kyledurand) [inline src/components/RangeSlider/components/SingleThumb/SingleThumb.scss:165]: Not a blocker since this will compile ok but we don't need to interpolate var() so `#{var()}` is unnecessary

File: src/tokens/_shape.ts
Comment (kyledurand) [inline src/tokens/_shape.ts:10]: I think this should still be:
```suggestion
  'border-radius-full': '50%',
```
or even 
```suggestion
  'border-radius-half': '50%',
```
I'll leave the naming up to you but I feel like it's part of a different group than the numbered values

File: src/components/DropZone/DropZone.scss
Comment (kyledurand) [inline src/components/DropZone/DropZone.scss:20]:
```suggestion
  border-radius: calc(var(--p-border-radius-1) + #{border-width()} + 1px);
```

File: src/components/IndexTable/IndexTable.scss
Comment (kyledurand) [inline src/components/IndexTable/IndexTable.scss:421]: I think we can 🔥 this custom property and use `var(--p-border-radius-1)` directly below

File: src/components/RangeSlider/components/SingleThumb/SingleThumb.scss
Comment (kyledurand) [inline src/components/RangeSlider/components/SingleThumb/SingleThumb.scss:65]:
```suggestion
      -1 * (#{$size} - var(--pc-range-slider-track-height)) / 2
```

File: src/components/RangeSlider/components/SingleThumb/SingleThumb.scss
Comment (kyledurand) [inline src/components/RangeSlider/components/SingleThumb/SingleThumb.scss:191]:
```suggestion
      var(--pc-range-slider-thumb-size) + #{$range-thumb-size-difference}
```

**Review by kyledurand — APPROVED** [review]
Code looks great. I'll check chromatic and approve there too if :gucci:

Feel free to address these comments in a follow up. Nothing blocking

**Comment by lgriffee** [comment]
@kyledurand Thank you so much for all your feedback and for helping me to troubleshoot the RangeSlider issues!! I'm going to address all your feedback in a follow up PR!

File: src/tokens/_shape.ts
Comment (lgriffee) [reply src/tokens/_shape.ts:10]: Yeah I was wondering about this—glad you brought it up! I like `'border-radius-half':` in case we have a 100% value at some point.

**Pull request merged**
